### PR TITLE
Added checking for Strata names

### PR DESCRIPTION
**Pull request by aklochkova**
Fixes https://github.com/OHDSI/Atlas/issues/1291

### CODE REVIEW

File: js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.html [hunk 1]
```diff
@@ -36,12 +36,14 @@ <h2 data-bind="css: classes('strata-title')">Stratas</h2>
         <div data-bind="foreach: stratas">
             <div data-bind="css: $component.classes('strata')">
                 <div data-bind="css: $component.classes('strata-heading')">
-                    <input type="text" data-bind="css: $component.classes({element: 'strata-name', extra: 'form-control'}), value: $data.name" placeholder="Strata name">
-                    <button
-                            type="button"
-                            data-bind="css: $component.classes({ element: 'strata-delete', extra: 'btn btn-sm btn-danger' }), click: () => $component.removeStrata($index())"
+                    <input type="text" data-bind="css: $component.getClassName($data.name() == ''), textInput: $data.name" placeholder="Strata name">
```
Comment (pavgra) [inline js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.html]: Why cannot simply do
`classes({element: 'strata-name', extra: 'form-control ' + (isNameEmpty ? 'emptyInput' : '') })`
?

File: js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.js [hunk 2]
```diff
@@ -132,6 +133,15 @@ define([
                 return `<a href='#' data-bind="click: () => $component.params.${action}('${d[identifierField]}')">Remove</a>`;
             }
         }
+        
+        getClassName(isNameEmpty){
+            this.disableSave(isNameEmpty);
```
Comment (pavgra) [inline js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.js]: `getClassName` which changes observable??

add computed based on design and updated passed variable via subscription

File: js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.js [hunk 3]
```diff
@@ -117,8 +118,18 @@ define([
...
             this.isParameterCreateModalShown = ko.observable(false);
-					  this.showConceptSetBrowser = ko.observable(false);
-					  this.criteriaContext = ko.observable();
+            this.showConceptSetBrowser = ko.observable(false);
+            this.criteriaContext = ko.observable();
+        }
+
+        checkStrataNames(data, event) {
+            let emptyNamesAmount = 0;
+            this.stratas().forEach(strata => {
```
Comment (johnSamilin) [inline js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.js]: use `.filter` or `.find`

File: js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.html [hunk 4]
```diff
@@ -36,12 +36,15 @@ <h2 data-bind="css: classes('strata-title')">Stratas</h2>
...
-                    <input type="text" data-bind="css: $component.classes({element: 'strata-name', extra: 'form-control'}), value: $data.name" placeholder="Strata name">
-                    <button
-                            type="button"
-                            data-bind="css: $component.classes({ element: 'strata-delete', extra: 'btn btn-sm btn-danger' }), click: () => $component.removeStrata($index())"
+                    <input type="text" data-bind="css: $component.classes({element: 'strata-name', extra: 'form-control' + ($data.name() == '' ? ' emptyInput' : '')}), 
+                                 textInput: $data.name, event: {'keyup': $component.checkStrataName}" placeholder="Strata name">
+                    <button type="button" data-bind="css: $component.classes({ element: 'strata-delete', extra: 'btn btn-sm btn-danger' }), click: () => $component.removeStrata($index())"
                     ><i class="fa fa-trash-o"></i></button>
                 </div>
+                <div data-bind="visible: $data.name() == ''" class="empty-name-error">
```
Comment (johnSamilin) [inline js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-design.html]: I'd use `css: { 'empty-name-error': !$data.name() }` binding

**Pull request merged**